### ISSUE TITLE
Bizzare block argument de-structuring issue on 2.5.8

### ISSUE THREAD

**Issue by glebm** [rubinius/rubinius#3522]
I am hitting a bizzare de-structuring issue.

When doing a de-structuring like this one:

```ruby
[['x', {'a' => 1}]].map { |(fst, snd)| ... }
```

`fst` is `{}`, and `snd` is `nil`!

However, when trying to reproduce it with a simple test-case, I couldn't, but it can be reproduced within a complex program. Steps to reproduce:

```sh
git clone https://github.com/glebm/i18n-tasks
git checkout f7137be6a8ce47838dd33a101364cdc0f196c0b1
bundle
bundle exec i18n-tasks health # <- boom!
```

* [The line with the de-structuring](https://github.com/glebm/i18n-tasks/blob/f7137be6a8ce47838dd33a101364cdc0f196c0b1/lib/i18n/tasks/used_keys.rb#L53).
* [Failing job on travis](https://travis-ci.org/glebm/glebm/i18n-tasks/jobs/87610980).

**Comment by glebm** [comment]
Any kind soul wants to look into this?

**Comment by ahmadsherif** [comment]
I think this is the same as #3544 (or the other way around, you reported it first). I've proposed a fix for it [here](https://github.com/rubinius/rubinius-ast/pull/6), but since @brixen is in the process of moving the code of that repo (and other ones) to [rubinius-code](https://github.com/rubinius/rubinius-code), it's taking some time to be reviewed.

**Comment by ahmadsherif** [comment]
Fixed in v3.14.
